src/core/qgsfeatureiterator.h

```diff
@@ -89,6 +89,9 @@ class CORE_EXPORT QgsAbstractFeatureIterator
 
     //! Setup the simplification of geometries to fetch using the specified simplify method
     virtual bool prepareSimplification( const QgsSimplifyMethod& simplifyMethod );
+    //! Prepare order by 
+    virtual void prepareOrderBy( const QStringList& orderByColumns );
```

Comment:
I think you'll need something more flexible then a string list here - specifically something which can also handle whether the field should be sorted ascending or descending